walletnode/services/artworkregister/state/state_test.go

```diff
@@ -0,0 +1,185 @@
+					newTestStatus("2020-05-01 00:01:00", StatusTaskCompleted, true),
+				},
+			},
+			expected: []*Status{
+				newTestStatus("2020-05-01 00:00:00", StatusTaskStarted, false),
+				newTestStatus("2020-05-01 00:00:01", StatusTicketAccepted, false),
+				newTestStatus("2020-05-01 00:01:00", StatusTaskCompleted, true),
+			},
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
+
+		for i, testCase := range testCases {
+
+			testCase := testCase
+
+			t.Run(fmt.Sprintf("test-case:%d", i), func(t *testing.T) {
+				t.Parallel()
+
+				states := testCase.state.All()
+				assert.Equal(t, testCase.expected, states)
+			})
+
+		}
+
+	})
+
+}
+
+func TestStateLatest(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		state  *State
+		status *Status
+	}{
+		{
+			state: &State{
+				statuses: []*Status{
+					newTestStatus("2020-01-02 01:00:00", StatusTaskStarted, false),
+					newTestStatus("2020-01-02 01:00:02", StatusTaskCompleted, true),
+				},
+			},
+			status: newTestStatus("2020-01-02 01:00:02", StatusTaskCompleted, true),
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
+
+		for i, testCase := range testCases {
+			testCase := testCase
+
+			t.Run(fmt.Sprintf("test-case:%d", i), func(t *testing.T) {
+				t.Parallel()
+
+				latest := testCase.state.Latest()
+				assert.Equal(t, testCase.status, latest)
+			})
+		}
+
+	})
+
+}
+
+func TestStateSubscribe(t *testing.T) {
+	t.Parallel()
+
+	testCases := []*Status{
+		newTestStatus("2020-01-02 00:00:00", StatusTaskStarted, false),
+		newTestStatus("2020-01-02 00:00:05", StatusConnected, false),
+		newTestStatus("2020-01-02 00:00:10", StatusTaskCompleted, true),
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
+
+		states := &State{
+			statuses: []*Status{
+				newTestStatus("2020-01-02 00:00:00", StatusTaskStarted, false),
+				newTestStatus("2020-01-02 00:00:05", StatusConnected, false),
+				newTestStatus("2020-01-02 00:00:10", StatusTaskCompleted, true),
+			},
+		}
+
+		sub, err := states.Subscribe()
+		assert.NoError(t, err)
+
+		for i, testCase := range testCases {
+			testCase := testCase
+
+			//no need run in parallel as only check status value
+			t.Run(fmt.Sprintf("test-case:%d", i), func(t *testing.T) {
+
+				status := <-sub.statusCh
+				assert.Equal(t, testCase, status)
+			})
+		}
+	})
+
+}
+
+func TestStateUpdate(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		states *State
+		status *Status
+	}{
+		{
+			states: &State{
+				statuses: []*Status{
+					newTestStatus("2021-01-01 00:00:00", StatusTaskStarted, false),
+				},
+			},
+			status: newTestStatus("2021-01-01 00:00:10", StatusTaskRejected, true),
+		},
+		{
+			states: &State{
+				statuses: []*Status{
+					newTestStatus("2021-01-01 00:00:00", StatusTaskStarted, false),
+				},
+			},
+			status: newTestStatus("2021-01-01 00:00:10", StatusTaskCompleted, true),
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
```

Comment:
```suggestion
```
`group` don't need to use `t.Parallel()`.
Actually we can remove  Run `group` at all from all places. It's only needed when we run some sub service, and we should stop it only when all testCases will be finished.

I also remember that I insisted to include the `group`, so it is my fault. But if you want to understand better how it works, you can re-read one more time this [article](https://gruntwork.io/guides/style%20guides/golang-style-guide/#testing) 
pay attention to
```go
    defer tearDownVPC()
    deployVPC()
``` 

> // Wrap the parallel tests in a synchronous test group to
// ensure that the main test function (the one calling
// `tearDownVPC` and `deployVPC`) waits until all the
// subtests are done before running the deferred function.

Reply:
ok. I am reading again. thank you